### PR TITLE
Enhancement/staging sadv2 17-23/07/2017

### PR DESCRIPTION
**Pull request by Wardormeur**
Sorry @DanielBrierton

### CODE REVIEW

File: lib/validations/dojos.js [hunk 1]
```diff
@@ -62,10 +69,8 @@ module.exports = function () {
         parentName: Joi.string(),
         address: Joi.string(),
         phone: joiValidator.phone(),
-        twitter: joiValidator.twitter().allow('').allow(null),
-        // What about facebook?
-        // facebook: joiValidator.facebook(),
```
Comment (butlerx) [inline lib/validations/dojos.js]: what about facebook?

File: web/public/js/directives/cd-start-dojo/champion/index.js [hunk 2]
```diff
@@ -70,6 +70,22 @@ angular
           ctrl.getAge();
           ctrl.formatDoB();
         }, true);
+
+        var phoneWatcher = $scope.$watch('$ctrl.champion.phone', function () {
+          if (ctrl.champion && ctrl.champion.phone) {
+            $timeout(function () {
+              // Force reload of value with async loading of data
+              var element = $('input[name="phone"]');
```
Comment (DanielBrierton) [inline web/public/js/directives/cd-start-dojo/champion/index.js]: Could inject `$element` and use that to find the input to scope it to the directive.

File: web/public/js/directives/cd-start-dojo/champion/template.dust [hunk 3]
```diff
@@ -74,23 +74,35 @@
       <label for="phone">{@i18n key="Phone number"/}</label>
       <input type="tel" name="phone" id="phone" autocomplete="off" class="form-control"
         utils-script="/components/intl-tel-input/lib/libphonenumber/build/utils.js"
-        required required-message="'{@i18n key="Telephone Number is empty"/}'"
-        ng-enter="noop()" international-phone-number ng-model="$ctrl.champion.phone">
+         required cd-phone-validator international-phone-number ng-model="$ctrl.champion.phone">
+        <div ng-messages="$ctrl.championForm.phone.$error" ng-show="$ctrl.championForm.phone.$touched">
+          <label ng-show="!$ctrl.championForm.phone.$valid" class="control-label has-error validationMessage">{@i18n key="Telephone number is not valid"/}</label>
```
Comment (DanielBrierton) [inline web/public/js/directives/cd-start-dojo/champion/template.dust]: Why a second ng-show? Could just combine with the one in the div above.

File: web/public/js/directives/cd-start-dojo/champion/template.dust [hunk 4]
```diff
@@ -74,23 +74,35 @@
...
       </div>
       <div class="form-group cd-simple-flex-grid__flex-1">
         <label for="phone">{@i18n key="LinkedIn"/}</label>
         <span class="pull-right">{@i18n key="(optional)"/}</span>
-        <input type="text" class="form-control" ng-model="$ctrl.champion.linkedin"
-          cd-input-prefix="https://www.linkedin.com/in/"
-          name="linkedin" id="linkedin" placeholder="{@i18n key="e.g. https://www.linkedin.com/in/SuperMario"/}">
+        <span class="input-group">
+          <span class="input-group-addon">
+            https://www.linkedin.com/
```
Comment (DanielBrierton) [inline web/public/js/directives/cd-start-dojo/champion/template.dust]: `https://www.linkedin.com/in/`

File: web/public/js/directives/cd-start-dojo/index.js [hunk 5]
```diff
@@ -148,13 +148,13 @@ angular
           var currentStep = _.find(ctrl.tabs, {state: $state.current.name});
           _.each(application, function (step, key, application) {
             if (currentStep.name === key) { // We modify only the current step
-              application[key].isValid = step.isValid ? step.isValid : (!_.isUndefined(step.formValidity) ? step.formValidity : false) ;
+              application[key].isValid = !_.isUndefined(step.formValidity) ? step.formValidity : (step.isValid ? step.isValid  : false) ;
```
Comment (DanielBrierton) [inline web/public/js/directives/cd-start-dojo/index.js]: Ternary operators inside ternary operators 🤢

File: web/public/js/directives/cd-start-dojo/index.js [hunk 6]
```diff
@@ -148,13 +148,13 @@ angular
...
               // Flag to display missing field : it's a returning user to this page
               application[key].visited = true;
             }
             delete step.form;
             delete step.formValidity;
-            application[key] = _.omitBy(step, _.isNil); // Remove null keys to avoid validations errors
+            application[key] = _.omitBy(step, function (val) {return _.isNil(val) || val === '';}); // Remove null keys to avoid validations errors
```
Comment (DanielBrierton) [inline web/public/js/directives/cd-start-dojo/index.js]: <3 being able to use arrow functions in Vue :P

File: web/public/js/directives/dojo/manage/template.dust [hunk 7]
```diff
@@ -108,6 +108,9 @@
                 <span ng-click="$ctrl.toggleSort($event, 'completedAt')"
                  ng-class="$ctrl.getSortClass(sort.completedAt)">&nbsp;</span>{@i18n key="Submitted at"/}
               </span>
+              <span ng-if="$ctrl.filter.verified !== 0 && $ctrl.filter.verified !== 1" >
```
Comment (DanielBrierton) [inline web/public/js/directives/dojo/manage/template.dust]: Show if it's simultaneously verified and not verified..?

File: web/public/js/services/dojo-utils.js [hunk 8]
```diff
@@ -95,5 +95,40 @@ angular.module('cpZenPlatform').factory('dojoUtils', ['$location', '$translate',
...
+      { id: '1/m',
+      name: $translate.instant('Monthly')},
+      { id: 'other',
+      name: $translate.instant('Other')}
+    ];
+    strings.monthlyFrequencies = [
+      {id: 'first', name: $translate.instant('First')},
+      {id: '2nd', name: $translate.instant('Second')},
+      {id: '3rd', name: $translate.instant('Third')},
+      {id: 'last', name: $translate.instant('Last')}
```
Comment (DanielBrierton) [inline web/public/js/services/dojo-utils.js]: 4th should surely be here too? A Dojo could run the 4th Monday of each month whether the month has 4 or 5 Mondays. Last then would be the 4th or 5th Monday of the month, depending on the month.

File: lib/dojos.js [hunk 9]
```diff
@@ -283,11 +233,39 @@ exports.register = function (server, options, next) {
...
         payload: Joi.object({
-          verified: Joi.boolean()
+          verified: Joi.number().valid(0).valid(1)
         })
       }
     }
-  }, {
+  },
+  {
+    method: 'PATCH',
```
Comment (DanielBrierton) [inline lib/dojos.js]: POST probably makes more sense for this, but meh, I could be wrong.

File: lib/dojos.js [hunk 10]
```diff
@@ -283,11 +233,39 @@ exports.register = function (server, options, next) {
           id: Joi.string().guid().required()
         },
         payload: Joi.object({
-          verified: Joi.boolean()
+          verified: Joi.number().valid(0).valid(1)
```
Comment (Wardormeur) [inline lib/dojos.js]: I'm still confused because the b-end only takes a smallint

File: web/public/js/directives/cd-sidebar/style.less [hunk 11]
```diff
@@ -103,6 +101,12 @@
 
     &__tabs {
       border-right: none;
+      &--deployable {
```
Comment (Wardormeur) [inline web/public/js/directives/cd-sidebar/style.less]: expandable ?

File: web/public/js/directives/dojo/manage/template.dust (hunk 7 above)
Comment (Wardormeur) [inline web/public/js/directives/dojo/manage/template.dust]: Yes, it's a generic title when we remove the filter on "verified"

File: web/public/js/directives/cd-sidebar/style.less (hunk 11 above)
Comment (DanielBrierton) [inline web/public/js/directives/cd-sidebar/style.less]: Oh, it's when you can hide the sidebar?

Collapsible.

File: web/public/js/services/dojo-utils.js (hunk 8 above)
Comment (Wardormeur) [inline web/public/js/services/dojo-utils.js]: My point is that having a 5th week is so "unexpected" that you plan on the scenario that is consistent

**Pull request merged**